src/dex/nerve/nerve.ts

```diff
@@ -0,0 +1,405 @@
+    blockNumber?: number,
+  ): Promise<DeepReadonly<{ state: PoolState; pool: EventPoolOrMetapool }[]>> {
+    const _pools = pools === undefined ? this.allPools : pools;
+
+    const _blockNumber =
+      blockNumber === undefined
+        ? await this.dexHelper.provider.getBlockNumber()
+        : blockNumber;
+
+    return Promise.all(
+      _pools.map(async eventPool => {
+        let state = eventPool.getState(_blockNumber);
+        if (!state || !state.isValid) {
+          this.logger.info(
+            `State for ${this.dexKey} pool ${eventPool.name} is stale or invalid on block ${_blockNumber}. Generating new one`,
+          );
+          const newState = await eventPool.generateState(_blockNumber);
+          eventPool.setState(newState, _blockNumber);
+          return { state: newState, pool: eventPool };
+        } else {
+          return { state, pool: eventPool };
+        }
+      }),
+    );
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    return this.allPools
+      .filter(pool => {
+        return (
+          pool.tokenAddresses.includes(srcToken.address.toLowerCase()) &&
+          pool.tokenAddresses.includes(destToken.address.toLowerCase())
+        );
+      })
+      .map(pool => Nerve.getIdentifier(this.dexKey, pool.address));
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<NerveData>> {
+    try {
+      if (side === SwapSide.BUY) return null;
+
+      const _srcToken = wrapETH(srcToken, this.network);
+      const _destToken = wrapETH(destToken, this.network);
+
+      if (
+        _srcToken.address.toLowerCase() === _destToken.address.toLowerCase()
+      ) {
+        return null;
+      }
+
+      const filterPoolsByIdentifiers = (
+        identifiers: string[],
+        pools: EventPoolOrMetapool[],
+      ) => {
+        return pools.filter(pool =>
+          identifiers.includes(Nerve.getIdentifier(this.dexKey, pool.address)),
+        );
+      };
+
+      const selectedPools = !limitPools
+        ? filterPoolsByIdentifiers(
+            await this.getPoolIdentifiers(
+              _srcToken,
+              _destToken,
+              side,
+              blockNumber,
+            ),
+            this.allPools,
+          )
+        : filterPoolsByIdentifiers(limitPools, this.allPools);
+
+      const statePoolPair = await this.getStates(selectedPools, blockNumber);
+
+      // here side === SwapSide.SELL
+      const unitVolume = getBigIntPow(10 ** _srcToken.decimals);
+      const chunks = amounts.length - 1;
+
+      const _width = Math.floor(chunks / NERVE_CHUNKS);
+      const _amounts = [unitVolume].concat(
+        Array.from(Array(NERVE_CHUNKS).keys()).map(
+          i => amounts[(i + 1) * _width],
+        ),
+      );
```

Comment:
Do we need to use lower chunks for Nerve? It should be fine to just to pricing for all the volumes

Reply:
Usually lower chunks are required if you RPC or if the computation is very heavy. With event systems its usually better to get prices for all the volumes

Reply:
I actually didn't understand this chunking scheme and reimplemented the one which was originally used. Thanks for clarifying :)
Fixed